src/org/labkey/test/components/ui/search/ManageSampleFinderViewsModal.java

```diff
@@ -0,0 +1,158 @@
+package org.labkey.test.components.ui.search;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ManageSampleFinderViewsModal extends ModalDialog
+{
+    public ManageSampleFinderViewsModal(WebDriver driver)
+    {
+        this("Manage Saved Searches", driver);
+    }
+
+    protected ManageSampleFinderViewsModal(String title, WebDriver driver)
+    {
+        super(new ModalDialog.ModalDialogFinder(driver).withTitle(title));
+    }
+
+    public void editViewName(String viewName)
+    {
+        WebElement editIcon = getViewEditIcon(viewName);
+        if (editIcon != null)
+        {
+            editIcon.click();
+        }
+        clearElementCache();
+        elementCache(); // waitForReady
```

Comment:
I'm not sure you need to do this. If the concern is that the controls will become stale/invalid you could try using a refindWhenNeeded for the controls in the element cache. The refind will find the elements every time they are referenced.

Alternatively if you are waiting for the dialog to be ready (it takes time to get and render the data) you could try implementing a waitForReady method that you explicitly call. There is an example of one in org.labkey.test.components.inventory.shared.FMBaseLocationTreeDialog